packages/devtools/lib/src/timeline/timeline_protocol.dart

```diff
@@ -88,40 +92,69 @@ class TimelineData {
 
   void _handleFrameStartEvent(TraceEvent event) {
     if (event.id != null) {
-      if (_pendingFrames[event.id] == null) {
+      final String id = _getFrameId(event);
```

Comment:
not really related to your CL but a quick cleanup of this code would be to write.
var pendingFrame = _pendingFrames.putIfAbsent(id, () => TimelineFrame(id);
// make rest of the logic just in terms of pendingFrame instead of pendingFrame[id].

Reply:
done.